plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesFunction.java

```diff
@@ -0,0 +1,186 @@
+                        ScalarArgumentSpecification.builder().name(TABLE_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(SINCE_VERSION_ARGUMENT).type(BIGINT).defaultValue(-1L).build()),
+                GENERIC_TABLE);
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+        this.deltaLakeMetadataFactory = requireNonNull(deltaLakeMetadataFactory, "deltaLakeMetadataFactory is null");
+    }
+
+    @Override
+    public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+    {
+        ScalarArgument schemaNameArgument = (ScalarArgument) arguments.get(SCHEMA_NAME_ARGUMENT);
+        checkFunctionArgument(schemaNameArgument.getValue() != null, "schema_name value for function table_changes() cannot be null");
+        String schemaName = ((Slice) schemaNameArgument.getValue()).toStringUtf8();
+
+        ScalarArgument tableNameArgument = (ScalarArgument) arguments.get(TABLE_NAME_ARGUMENT);
+        checkFunctionArgument(tableNameArgument.getValue() != null, "table_name value for function table_changes() cannot be null");
+        String tableName = ((Slice) tableNameArgument.getValue()).toStringUtf8();
+
+        ScalarArgument sinceVersionArgument = (ScalarArgument) arguments.get(SINCE_VERSION_ARGUMENT);
+        if (sinceVersionArgument.getValue() == null) {
+            throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "since_version argument must be a number, null provided");
+        }
+        long sinceVersion = (long) sinceVersionArgument.getValue() + 1; //+1 to ensure that the sinceVersion is exclusive
+        checkFunctionArgument(sinceVersion >= 0, "since_version value for function table_changes() cannot be negative");
+
+        DeltaLakeMetadata deltaLakeMetadata = deltaLakeMetadataFactory.create(session.getIdentity());
+        SchemaTableName schemaTableName = new SchemaTableName(schemaName, tableName);
+        ConnectorTableHandle connectorTableHandle = deltaLakeMetadata.getTableHandle(session, schemaTableName);
+        if (connectorTableHandle instanceof CorruptedDeltaLakeTableHandle corruptedTableHandle) {
+            throw corruptedTableHandle.createException();
+        }
+        DeltaLakeTableHandle tableHandle = (DeltaLakeTableHandle) connectorTableHandle;
+        if (tableHandle == null) {
+            throw new TableNotFoundException(schemaTableName);
+        }
+        validateThatCdfWasEnabled(session, fileSystemFactory, tableHandle, sinceVersion);
+        List<DeltaLakeColumnHandle> columnHandles = deltaLakeMetadata.getColumnHandles(session, tableHandle)
+                .values().stream()
+                .map(DeltaLakeColumnHandle.class::cast)
+                .filter(column -> !IGNORED_COLUMNS.contains(column.getName()))
```

Comment:
```suggestion
                         .filter(column -> column.getColumnType() != SYNTHESIZED)
```

It seems more future proof.

Please remove `IGNORED_COLUMNS` if you take this suggestion.